core/src/main/java/org/opensearch/sql/calcite/CalciteRexNodeVisitor.java

```diff
@@ -358,14 +360,21 @@ private RelNode resolveSubqueryPlan(UnresolvedPlan subquery, CalcitePlanContext
     return subqueryRel;
   }
 
-  /*
-   * Unsupported Expressions of PPL with Calcite for OpenSearch 3.0.0-beta
-   */
   @Override
   public RexNode visitCast(Cast node, CalcitePlanContext context) {
-    throw new CalciteUnsupportedException("CastWhen function is unsupported in Calcite");
+    RexNode expr = analyze(node.getExpression(), context);
+    SqlTypeName sqlTypeName =
+        OpenSearchTypeFactory.convertExprTypeToRelDataType(node.getDataType().getCoreType())
+            .getSqlTypeName();
+    RelDataType type = context.rexBuilder.getTypeFactory().createSqlType(sqlTypeName);
+    RelDataType nullableType =
+        context.rexBuilder.getTypeFactory().createTypeWithNullability(type, true);
+    return context.rexBuilder.makeCast(SqlParserPos.ZERO, nullableType, expr, false, true);
```

Comment:
use this 4 lines instead of `context.relbuilder.cast()` because we need the last parameter to `true`

Reply:
does Calcite has same [Data Type Conversion rule](https://github.com/opensearch-project/sql/blob/main/docs/user/general/datatypes.rst#id4)? anything we want to call-out and update doc?

Reply:
I will add more test for this table